### PR TITLE
Fix HyperV test cases against Ubuntu 20.04 (focal)

### PR DESCRIPTION
**Pull request by LiliDeng**
In Ubuntu 20.04 (focal), it used netplan to manage network. Before fix, the network related cases which need extra configuration failed.

Test results -
```
[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 13:31:22
VHD Under Test        : ubuntu_20.04.vhdx
Initial Kernel Version: 5.4.0-21-generic
Final Kernel Version  : 5.4.0-21-generic
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minut
es) 
---------------------------------------------------------------------------------------------------------------------------------------
----
    1 NETWORK              ETHTOOL-CHECK-STATISTICS                                                          PASS                 10.1 

[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 08:14:17
VHD Under Test        : ubuntu_20.04.vhdx
Initial Kernel Version: 5.4.0-21-generic
Final Kernel Version  : 5.4.0-21-generic
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:10

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU                                           PASS                 8.62 

[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 07:59:25
VHD Under Test        : ubuntu_20.04.vhdx
Initial Kernel Version: 5.4.0-21-generic
Final Kernel Version  : 5.4.0-21-generic
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-JUMBO-FRAMES                                                                  PASS                 5.46 

[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 07:55:01
VHD Under Test        : ubuntu_20.04.vhdx
Initial Kernel Version: 5.4.0-21-generic
Final Kernel Version  : 5.4.0-21-generic
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:4

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-EXTERNAL                                                                      PASS                 2.68 
```

Regression testing -
```
[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 08:49:28
VHD Under Test        : ubuntu_18.04.1_gen1_gen2.vhdx
Initial Kernel Version: 4.15.0-38-generic
Final Kernel Version  : 5.0.0-1036-azure
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-EXTERNAL                                                                      PASS                 2.42 

[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 10:10:59
VHD Under Test        : Ubuntu_18.04.1.vhdx
Initial Kernel Version: 4.15.0-38-generic
Final Kernel Version  : 5.4.0-rc8-5d1131b4d61e
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:20

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                ETHTOOL-OFFLOADING-SETTING                                                        PASS                 6.65 

[LISAv2 Test Results Summary]
Test Run On           : 04/14/2020 08:29:05
VHD Under Test        : ubuntu_18.04.1_gen1_gen2.vhdx
Initial Kernel Version: 4.15.0-38-generic
Final Kernel Version  : 5.0.0-1036-azure
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:18

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU                                           PASS                12.34 
```

### CODE REVIEW

File: Testscripts/Linux/utils.sh [hunk 1]
```diff
@@ -1357,47 +1368,59 @@ function CreateIfupConfigFile() {
...
+				if [ -d /etc/netplan/ ]; then
+					__file_path="/etc/netplan/01-static-network.yaml"
+					rm -rf $__file_path
+					echo "network:" >> "$__file_path"
+					echo "    version: 2" >> "$__file_path"
+					echo "    ethernets:" >> "$__file_path"
+					echo "        $__interface_name:" >> "$__file_path"
+					echo "            dhcp4: no" >> "$__file_path"
+					echo "            addresses: [$__ip/24]" >> "$__file_path"
+					netplan apply
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/utils.sh:1386]: Please add a log line indicating netplan applying to the user.

File: Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh [hunk 2]
```diff
@@ -133,7 +133,7 @@ if [ $? -ne 0 ]; then
 fi
 
 # Install iPerf3
-ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$VF_IP2" ". $HOME/utils.sh && update_repos && install_package iperf3"
+ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$VF_IP2" ". /home/${SUDO_USER}/utils.sh && update_repos && install_package iperf3"
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh:136]: I don't understand this change here. What is your purpose?

**Comment by lubaihua33** [comment]
Do we need to test against Ubuntu older version?

File: Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh (hunk 2 above)
Comment (LiliDeng) [reply Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh:136]: In most situation, this case is skipped, didn't run this logic at all. 
During testing, when it run install iperf3, the utils.sh file only exist under /home/$user folder.

Comment (LiliDeng) [reply Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh:136]: > I don't understand this change here. What is your purpose?

In most situation, this case is skipped, didn't run this logic at all. 
During testing, when it run install iperf3, the utils.sh file only exist under /home/$user folder.

**Comment by LiliDeng** [comment]
> Do we need to test against Ubuntu older version?

I already include them in the regression testing, please double check.

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**